server/sonar-server/src/main/java/org/sonar/server/startup/GeneratePluginIndex.java

```diff
@@ -45,22 +50,30 @@ public GeneratePluginIndex(ServerFileSystem fileSystem, PluginRepository reposit
     this.repository = repository;
   }
 
-  public void start() throws IOException {
+  @Override
+  public void start() {
+    Profiler profiler = Profiler.create(LOG).startInfo("Generate scanner plugin index");
     writeIndex(fileSystem.getPluginIndex());
+    profiler.stopDebug();
+  }
+
+  @Override
+  public void stop() {
+    // Nothing to do
   }
 
-  void writeIndex(File indexFile) throws IOException {
-    FileUtils.forceMkdir(indexFile.getParentFile());
-    Writer writer = new OutputStreamWriter(new FileOutputStream(indexFile), StandardCharsets.UTF_8);
+  void writeIndex(File indexFile) {
     try {
-      for (PluginInfo pluginInfo : repository.getPluginInfos()) {
-        writer.append(RemotePlugin.create(pluginInfo).marshal());
-        writer.append(CharUtils.LF);
+      FileUtils.forceMkdir(indexFile.getParentFile());
+      try (Writer writer = new OutputStreamWriter(new FileOutputStream(indexFile), StandardCharsets.UTF_8)) {
+        for (PluginInfo pluginInfo : repository.getPluginInfos()) {
+          writer.append(RemotePlugin.create(pluginInfo).marshal());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "RemotePlugin"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)